packages/shared/lib/services/environment.service.ts

```diff
@@ -20,11 +20,16 @@ interface EnvironmentAccount {
 type EnvironmentAccountSecrets = Record<string, EnvironmentAccount>;
 
 export const defaultEnvironments = ['prod', 'dev'];
+const CACHE_ENABLED = !(process.env['NANGO_CACHE_ENV_KEYS'] === 'false');
```

Comment:
we have a feature flag util (reading keys from redis)
```
const useCache = await featureFlags.isEnabled('use-env-secret-cache', 'global', true);
```
If you want to be able to enable/disable it live you would need to keep adding to the cache and check the flag only when retrieving the data https://github.com/NangoHQ/nango/pull/1908/files#diff-f3d9e285186619d6dfde44bdbbb720119863b18a3a80a41025f4a9c83e13705aL97

Reply:
Indeed, I wasn't aware of this feature.
I think, reading this flag from redis at each call would === caching the secrets to Redis, defeating the purpose of this :D I'd leave it like this unless you really find it better

Reply:
If you are worried about skewing the telemetry. You can check the flag outside of the tracking and pass the boolean to `getAccountIdAndEnvironmentIdBySecretKey`. Up to you though